src/ui/components/PasswordModule/PasswordModule.test.tsx

```diff
@@ -0,0 +1,623 @@
+      expect(getByText(ENG_trans.createpassword.alert.text)).toBeVisible();
+    });
+
+    const mockDate = new Date(1466424490000);
+    const spy = jest
+      .spyOn(global, "Date")
+      .mockImplementation((() => mockDate) as never);
+    act(() => {
+      fireEvent.click(getByText(ENG_trans.createpassword.alert.button.confirm));
+    });
+
+    await waitFor(() => {
+      expect(createOrUpdateBasicRecordMock).toBeCalledWith(
+        new BasicRecord({
+          id: MiscRecordId.APP_PASSWORD_SKIPPED,
+          content: { value: true },
+        })
+      );
+    });
+
+    spy.mockRestore();
+  });
+
+  test("Submit password", async () => {
+    const { getByTestId } = render(
+      <Provider store={storeMocked(initialState)}>
+        <PasswordModule
+          title="Password Module"
+          description="Description"
+          testId="password-module"
+          isOnboarding={true}
+          onCreateSuccess={onCreateSuccesMock}
+        />
+      </Provider>
+    );
+
+    const input = getByTestId("create-password-input");
+    const confirmInput = getByTestId("confirm-password-input");
+    const hintInput = getByTestId("create-hint-input");
+
+    act(() => {
+      ionFireEvent.ionInput(input, "Passssssssss1@");
+      ionFireEvent.ionInput(confirmInput, "Passssssssss1@");
+      ionFireEvent.ionInput(hintInput, "hint");
+    });
+
+    await waitFor(() => {
+      expect(
+        getByTestId("password-validation-length-icon").classList.contains(
+          "pass"
+        )
+      ).toBe(true);
+      expect(
+        getByTestId("password-validation-uppercase-icon").classList.contains(
+          "pass"
+        )
+      ).toBe(true);
+      expect(
+        getByTestId("password-validation-lowercase-icon").classList.contains(
+          "pass"
+        )
+      ).toBe(true);
+      expect(
+        getByTestId("password-validation-number-icon").classList.contains(
+          "pass"
+        )
+      ).toBe(true);
+      expect(
+        getByTestId("password-validation-symbol-icon").classList.contains(
+          "pass"
+        )
+      ).toBe(true);
+    });
+
+    const submitButton = getByTestId("primary-button-password-module");
+
+    const mockDate = new Date(1466424490000);
+    const spy = jest
+      .spyOn(global, "Date")
+      .mockImplementation((() => mockDate) as never);
+
+    act(() => {
+      ionFireEvent.click(submitButton);
+    });
+
+    await waitFor(() => {
+      expect(secureStorageSetFunc).toBeCalledWith(
+        KeyStoreKeys.APP_OP_PASSWORD,
+        "Passssssssss1@"
+      );
+      expect(createOrUpdateBasicRecordMock).toBeCalledWith(
+        new BasicRecord({
+          id: MiscRecordId.OP_PASS_HINT,
+          content: { value: "hint" },
+        })
+      );
+    });
+
+    spy.mockRestore();
+  });
+
+  test("Submit password on manage password page", async () => {
+    const initialState = {
+      stateCache: {
+        routes: [RoutePath.GENERATE_SEED_PHRASE],
```

Comment:
`RoutePath.GENERATE_SEED_PHRASE` is the URL of Create Password during Onboarding (on Manage Password page the URL is `TabsRoutePath.MENU`).